### ISSUE TITLE
Functions deploy fails on firebase-tools version 13.15.3 but works in 13.15.2

### ISSUE THREAD

**Issue by Spitzbua** [firebase/firebase-tools#7575]
### [REQUIRED] Environment info

**firebase-tools:** 13.15.3

**Platform:** Windows

### [REQUIRED] Test case

Deploy firebase function v2

### [REQUIRED] Steps to reproduce

firebase deploy --only functions

### [REQUIRED] Expected behavior

Functions will be deployed

### [REQUIRED] Actual behavior

Deploy fails

> [debug] [2024-08-21T13:09:58.665Z] TypeError: Cannot convert undefined or null to object
    at Function.entries (<anonymous>)
    at Object.want (C:\Users\thoma\AppData\Roaming\fnm\node-versions\v20.16.0\installation\node_modules\firebase-tools\lib\deploy\extensions\planner.js:120:28)
    at prepareDynamicExtensions (C:\Users\thoma\AppData\Roaming\fnm\node-versions\v20.16.0\installation\node_modules\firebase-tools\lib\deploy\extensions\prepare.js:122:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async prepare (C:\Users\thoma\AppData\Roaming\fnm\node-versions\v20.16.0\installation\node_modules\firebase-tools\lib\deploy\functions\prepare.js:62:9)
    at async chain (C:\Users\thoma\AppData\Roaming\fnm\node-versions\v20.16.0\installation\node_modules\firebase-tools\lib\deploy\index.js:40:9)
    at async deploy (C:\Users\thoma\AppData\Roaming\fnm\node-versions\v20.16.0\installation\node_modules\firebase-tools\lib\deploy\index.js:97:5)
[error]

**Comment by aalej** [comment]
Hey @Spitzbua, thanks for reporting this! I was able to reproduce the error you mentioned. Let me raise this to our engineering team so they can take a look. For those also encountering this, could you try downgrading to v13.15.2 to workaround the issue.

**Comment by OlivierLebra** [comment]
Another workaround is to add an empty `"extensions": []` array field to your `firebase.json`.

**Comment by gregfenton** [comment]
I am seeing this issue pop up in 2 different public forums.  When deploying with `--debug` they are seeing:

```
[2024-08-21T17:09:37.182Z] <<< [apiv2][body] POST https://cloudresourcemanager.googleapis.com/v1/projects/jongekindspecialist-18d27:testIamPermissions {"permissions":["firebase.projects.get","firebaseextensions.instances.list"]}
[2024-08-21T17:09:37.237Z] TypeError: Cannot convert undefined or null to object
    at Function.entries (<anonymous>)
    at Object.want (C:\Users\FOOBAR\AppData\Roaming\npm\node_modules\firebase-tools\lib\deploy\extensions\planner.js:120:28)
    at prepareDynamicExtensions (C:\Users\FOOBAR\AppData\Roaming\npm\node_modules\firebase-tools\lib\deploy\extensions\prepare.js:122:48)
.......
.......
    ```

**Comment by dhamuace** [comment]
I am getting same issue in my end today.

![image](https://github.com/user-attachments/assets/c229a1c4-8b96-42c2-9ab7-7e1582d8dc93)

**Comment by dhamuace** [comment]
And guys after downgraded to v13.15.2 it finally works!!! Thanks for short workaround @aalej

**Comment by meisolated** [comment]
I was also facing the same issue, but making a small tweak to the command made it work for me.
![image](https://github.com/user-attachments/assets/c38859b8-6aef-447d-a482-2bd3c2b41f9e)

**Comment by joehan** [comment]
Hey all, sorry for the inconvenience here. This should be fixed in 13.15.4, which will be available on npm shortly

**Comment by gregfenton** [comment]
No need to be sorry. We've all been here and will be here a million times again.

See it as the opportunity to improve test automation that it is!  😁

**Comment by siarheidudko** [comment]
@joehan  Your fix has completely broken my project. 
In CI (Github Actions) I used the command when deploying:
`firebase deploy -m "Autodeploy from GitHUB ($GITHUB_ACTOR)" --only functions --force`

It has completely **removed all extensions and their settings**. Probably such actions lead to the fact that I will be forced to abandon firebase in the future, because your work is very unstable.

**Comment by siarheidudko** [comment]
@gregfenton I pay for Google services and I expect services of acceptable quality. In the concept of a service, I also include tools provided by Google or their subsidiaries such as Firebase. I understand mistakes when these mistakes do not lead to irreversible consequences. But today I had to spend several hours manually reinstalling and configuring all the deleted extensions. I also had to stop the project for a few hours while a decision was made to minimize the consequences.
The previous problem in 13.15.3, for example, did not lead to irreversible consequences and simply required a downgrade.
At the same time, the deployment of extensions from the documentation would have to occur with the commands `firebase deploy --only extensions` [link](https://firebase.google.com/docs/extensions/manifest?hl=en) But it doesn't work. 
Moreover, the description available in the console does not provide for deleting all extensions at once or installing them from the env file.
If you consider it appropriate for yourself to deploy a test environment only for testing firebase-tools and update your CI processes manually, then this is your personal business and your personal money.
